### PR TITLE
Expose flag for should_save_peft_only

### PR DESCRIPTION
**Pull request by irenedea**
This will allow users to control whether they want their composer checkpoints from lora to have the full weights or just the adapter weights.

Note: This only applies to composer checkpoints, not HF checkpoints. HF checkpoints saved through `hf_checkpointer.py` will still only contain the adapters only. 

## Tests
should_save_peft_only: False
```
2024-07-15 21:27:50     844110 .metadata
2024-07-15 21:27:55 4028659956 __0_0.distcp
2024-07-15 21:27:55 4019819348 __1_0.distcp
2024-07-15 21:27:55 4019819348 __2_0.distcp
2024-07-15 21:27:55 4019819348 __3_0.distcp
2024-07-15 21:27:55 4019819348 __4_0.distcp
2024-07-15 21:27:55 4019819348 __5_0.distcp
2024-07-15 21:27:55 4019819348 __6_0.distcp
2024-07-15 21:27:55 4019819348 __7_0.distcp
```

should_save_peft_only: True
```
2024-07-15 21:35:06     291838 .metadata
2024-07-15 21:35:06   13185952 __0_0.distcp
2024-07-15 21:35:06    4345344 __1_0.distcp
2024-07-15 21:35:06    4345344 __2_0.distcp
2024-07-15 21:35:06    4345344 __3_0.distcp
2024-07-15 21:35:06    4345344 __4_0.distcp
2024-07-15 21:35:06    4345344 __5_0.distcp
2024-07-15 21:35:06    4345344 __6_0.distcp
2024-07-15 21:35:06    4345344 __7_0.distcp
```

### CODE REVIEW

File: llmfoundry/models/hf/hf_causal_lm.py [hunk 1]
```diff
@@ -131,6 +132,7 @@ def __init__(
             eval_metrics=eval_metrics,
             init_device=init_device,
             peft_config=peft_config_object,
+            should_save_peft_only=should_save_peft_only,
```
Comment (snarayan21) [inline llmfoundry/models/hf/hf_causal_lm.py:135]: lol we didn't pass this flag in before? funny cuz it's on composer's HuggingFaceModel class

Comment (dakinggg) [reply llmfoundry/models/hf/hf_causal_lm.py:135]: didn't pass cause assumed everyone wanted just peft checkpoints

**Review by dakinggg — APPROVED** [review]

**Review by snarayan21 — CHANGES_REQUESTED** [review]
code changes lgtm.
before merging, mind doing a save/resumption test with and without the flag? want to make sure we can resume from checkpoints with this enabled

**Review by snarayan21 — APPROVED** [review]
lgtm

**Pull request merged**